### PR TITLE
Change static messages to dynamic comments from user-input form.

### PR DESCRIPTION
**Pull request by Sami-2000**
Add form to main page for user comments. Hold array of comments in server. Print comments to main page when reloading.

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -26,17 +26,24 @@
...
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String jsonComments = convertToJsonUsingGson(comments);
     ArrayList<String> messages = new ArrayList<String>();
     messages.add("Good morning.");
     messages.add("Good afternoon.");
     messages.add("Good evening.");
 
     String jsonMessages = convertToJsonUsingGson(messages);
```
Comment (josephdhurst) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:39]: It looks like this function doesn't modify the response at all, so I'm unsure what its purpose is. Correct me if I'm wrong, but I think that if you call this it won't do anything.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -26,17 +26,24 @@
...
+  }
 
-    response.setContentType("application/json;");
-    response.getWriter().println(jsonMessages);
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String comment = request.getParameter("text-input");
+    comments.add(comment);
+    response.sendRedirect("/index.html");
```
Comment (josephdhurst) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:46]: Why is this line necessary? Is it just refreshing index.html so that the new comment shows up on the page?

File: portfolio/src/main/webapp/script.js [hunk 3]
```diff
@@ -48,25 +48,24 @@ function bubbleFall() {
...
 
-    // Find container.
-    const messageContainer = document.getElementById('message-container');
-
-    // Add each message to container as list element.
-    for (let i = 0; i < 3; i++) {
-      messageContainer.appendChild(createListElement(messages[i]));
+    const commentSectionContainer = document.getElementById('comment-section-container');
+    
+    // Add each comment to container as comment element.
```
Comment (josephdhurst) [inline portfolio/src/main/webapp/script.js:58]: This comment is redundant because this for loop is simple enough and the function/method names are specific enough that I can understand it without comments. 

Here's a snippet from an internal doc that you may or may not have access to (https://engdoc.corp.google.com/eng/doc/devguide/java/practices/comments.md?cl=head#what_for):

What comments are for
Use a comment when it is infeasible to make your code self-explanatory. Here's a great quote from Robert C. Martin's Clean Code:

The proper use of comments is to compensate for our failure to express ourself in code. Note that I used the word failure. I meant it. Comments are always failures. We must have them because we cannot always figure out how to express ourselves without them, but their use is not a cause for celebration.... Every time you express yourself in code, you should pat yourself on the back. Every time you write a comment, you should grimace and feel the failure of your ability of expression.

Self-explanatory code beats liberally-commented code (which, of course, beats code that can't be understood). If you need a comment to explain what a block of code does, try to introduce an explaining variable or to extract a method. If you've already done that but you still need a comment, try to rename it until it answers all your concerns. Sometimes, just choosing better identifier names can be enough; sometimes it takes more work. It's almost always worth it.

File: portfolio/src/main/webapp/script.js [hunk 4]
```diff
@@ -48,25 +48,24 @@ function bubbleFall() {
...
+}S
 
-// Creates an <li> element containing text.
-// COPIED FROM SERVER-STATS EXAMPLE.
-function createListElement(text) {
-  const liElement = document.createElement('li');
-  liElement.innerText = text;
-  return liElement;
+// Creates an <comment> element containing text.
+function createCommentElement(text) {
```
Comment (josephdhurst) [inline portfolio/src/main/webapp/script.js:66]: From https://google.github.io/styleguide/jsguide.html#jsdoc-method-and-function-comments: In methods and named functions, parameter and return types must be documented

File: portfolio/src/main/webapp/script.js [hunk 5]
```diff
@@ -48,25 +48,24 @@ function bubbleFall() {
...
 }
 
-// Get messages from the data servlet.
-function getMessages() {
-  fetch('/data').then(response => response.json()).then((messages) => {
-    console.log(messages);
+// Get comments from the data servlet.
+function getComments() {
+  fetch('/data').then(response => response.json()).then((comments) => {
+    console.log(comments);
```
Comment (josephdhurst) [inline portfolio/src/main/webapp/script.js:54]: I'm assuming that comments is an `Array<string>` but I'm not completely sure because I know that json can take many different forms like a map. You might want to add parameter and return types here (but it's not mandatory).

https://google.github.io/styleguide/jsguide.html#jsdoc-method-and-function-comments

**Review by josephdhurst — APPROVED** [review]

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 2 above)
Comment (Sami-2000) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:46]: Yes! That was my own preference. Because the comments only appear when the page loads, I wanted the user to see their own comment right away.
